File: Back-End/Sabemi.Application/Features/Contracts/ContractModel.cs
using Sabemi.Domain.Enums;
namespace Sabemi.Application.Features.Contracts;

public class ContractModel
{
    public Guid Id { get; set; }
    public EContractStatus Status { get; set; }
    public string Name { get; set; }
    public decimal TotalAmount { get; set; }
    public decimal PaidAmount { get; set; }
    public decimal PendingAmount { get; set; }
    public DateTime UpdatedAt { get; set; }
    public DateTime CreatedAt { get; set; }
    public IEnumerable<ContractPaymentModel> Payments { get; set; } = [];
}

File: sabemi-front-app/src/models/contract.model.ts
import type { ContractStatus } from "../enums/contract.enum";

export interface Contract {
    id: string;
    status: ContractStatus;
    name: string;
    totalAmount: number;
    paidAmount: number;
    pendingAmount: number;
    updatedAt: Date;
    createdAt: Date;

    payments?: ContractPayment[];
}

export interface ContractPayment {
    id: string;
    contractId: string;
    transactionId: string;
    amount: number;
    paidAt: Date;
    createdAt: Date;
    contract?: Contract;
}